src/main/java/com/dnd/runus/application/badge/BadgeService.java

```diff
@@ -52,36 +57,46 @@ public AchievedBadgesResponse getAchievedBadges(long memberId) {
+        List<BadgeAchievement.OnlyBadge> allBadges =
+                badgeAchievementRepository.findByMemberIdOrderByBadgeTypeAndAchievedAt(memberId);
 
+        // 최신 획득한 뱃지(오늘 기준으로 일주일)
         LocalDateTime oneWeekAgo = LocalDate.now(SERVER_TIMEZONE_ID)
                 .atStartOfDay(SERVER_TIMEZONE_ID)
                 .toOffsetDateTime()
                 .minusDays(7)
                 .toLocalDateTime();
 
-        EnumMap<BadgeType, List<BadgeWithAchievedStatus>> badgeMap = allBadges.stream()
+        List<AchievedBadge> recencyBadges = allBadges.stream()
+                .filter(v -> oneWeekAgo.isBefore(v.createdAt().toLocalDateTime()))
+                .map(v -> new AchievedBadge(
+                        v.badge().badgeId(),
+                        v.badge().name(),
+                        v.badge().imageUrl(),
+                        v.createdAt().toLocalDateTime()))
+                .toList();
+
+        // badgeType별 획득한 배지
+        EnumMap<BadgeType, List<AchievedBadge>> badgesWithType = allBadges.stream()
                 .collect(Collectors.groupingBy(
                         v -> v.badge().type(),
                         () -> new EnumMap<>(BadgeType.class),
-                        Collectors.mapping(BadgeWithAchievedStatus::from, Collectors.toList())));
+                        Collectors.mapping(
+                                v -> new AchievedBadge(
+                                        v.badge().badgeId(),
+                                        v.badge().name(),
+                                        v.badge().imageUrl(),
+                                        v.createdAt().toLocalDateTime()),
+                                Collectors.toList())));
 
-        List<BadgeWithAchievedStatus> recencyBadges = allBadges.stream()
-                .filter(badge -> isRecent(badge, oneWeekAgo))
-                .map(BadgeWithAchievedStatus::from)
-                .toList();
+        // 타입 별, showPriority순서로 리스트에 추가
+        TreeSet<BadgeType> badgeTypesSet = new TreeSet<>(Comparator.comparingInt(BadgeType::getShowPriority));
+        badgeTypesSet.addAll(EnumSet.allOf(BadgeType.class));
 
-        return new AllBadgesListResponse(
-                recencyBadges,
-                badgeMap.getOrDefault(BadgeType.PERSONAL_RECORD, Collections.emptyList()),
-                badgeMap.getOrDefault(BadgeType.DISTANCE_METER, Collections.emptyList()),
-                badgeMap.getOrDefault(BadgeType.STREAK, Collections.emptyList()),
-                badgeMap.getOrDefault(BadgeType.DURATION_SECONDS, Collections.emptyList()),
-                badgeMap.getOrDefault(BadgeType.LEVEL, Collections.emptyList()));
-    }
+        List<BadgesWithType> responseBadges = new ArrayList<>();
+        badgeTypesSet.forEach(badgeType -> responseBadges.add(new BadgesWithType(
+                badgeType.getName(), badgesWithType.getOrDefault(badgeType, Collections.emptyList()))));
```

Comment:
외부 변수를 선언하고 forEach에서 스트림 외부의 값을 조작하기 보다는,
스트림 map을 사용해 리스트로 반환하면 좋을 것 같아요

```suggestion
        List<BadgesWithType> responseBadges = badgeTypesSet.stream()
                .map(badgeType -> new BadgesWithType(
                        badgeType.getName(), badgesWithType.getOrDefault(badgeType, Collections.emptyList())))
                .toList();
```

Reply:
자바 stream API 공부를 좀 더 해야겠네요😅